azurerm/data_source_batch_certificate.go

```diff
@@ -0,0 +1,76 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmBatchCertificate() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmBatchCertificateRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAzureRMBatchCertificateName,
+			},
+			"account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAzureRMBatchAccountName,
+			},
+			"resource_group_name": resourceGroupNameSchema(),
+			"public_data": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"format": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"thumbprint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"thumbprint_algorithm": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmBatchCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).batchCertificateClient
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+	accountName := d.Get("account_name").(string)
+	name := d.Get("name").(string)
+
+	ctx := meta.(*ArmClient).StopContext
```

Comment:
could we move this up to the top of the function for consistency with the other resources?

Reply:
Done. I mirrored the style in data_source_batch_account.go - do you want that updated to reflect the style as well?